doc/coq.rst

```diff
@@ -0,0 +1,507 @@
+.. _coq:
+
+***
+Coq
+***
+
+
+Dune is able to build Coq projects and plugins with support for source file
+preprocessing, extraction, and documenation.
+
+
+.. contents:: Table of Contents
+    :depth: 3
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+A *Coq theory* is a collection of ``.v`` files. Together with OCaml libraries
+linking to the Coq API, a Coq theory is termed a *Coq plugin*. Both may also be
```

Comment:
I think we can improve this a bit. A Coq theory is indeed a set of .v files in several directories, prefixed by a Coq Module name, and following the module hierachy that a directory sets.

A Coq plugin is just a regular OCaml library, and is unrelated to a Coq theory. A Coq theory may depend on a plugin.

I think it is not correct to say "A Coq theory is termed a Coq plugin".

A Coq project can be a collection of plugins and libraries, and other stuff indeed, but I'm not sure the term belongs here as it doesn't correspond to a notion more than actually a Coq project is really a Dune project.